### PR TITLE
Rebuild for pugixml114 redux

### PR DESCRIPTION
**Pull request by larsoner**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Trying to take over #11

Closes #11

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -2,5 +2,6 @@ cmake $CMAKE_ARGS \
 	-DCMAKE_BUILD_TYPE=Release \
 	-DLSL_BUNDLED_PUGIXML=OFF \
 	-DLSL_UNIXFOLDERS=ON \
+	-DCMAKE_CXX_STANDARD=11 \
```
Comment (larsoner) [inline recipe/build.sh:5]: Turns out this was the key change to fix osx x86_64

**Comment by larsoner** [comment]
@cboulay @tstenner should be good go merge here! (FYI I want this so that liblsl is compatible with VTK 9.3 that just came out, e.g. [here](https://github.com/mne-tools/mne-installers/pull/273)).

File: recipe/build.sh (hunk 1 above)
Comment (hoechenberger) [reply recipe/build.sh:5]: Great find!!

**Review by cboulay — APPROVED** [review]
I don't have much investment in the conda build. I looked through the commits and I didn't see anything alarming. If it works for you then that's good enough for me.

**Comment by larsoner** [comment]
Yeah it's really just two lines changed by me (stdlib line and c++11 line) the rest is from #11 so should be good to go

**Comment by tstenner** [comment]
IIRC 1.16.2 uses C++14 features, but maybe it was just C++11 feature that were not fully supported by the compilers at the time. Then again I think it is overridden in the `CMakeLists.txt`, so it shouldn't break anything.

**Review by tstenner — APPROVED** [review]

**Comment by larsoner** [comment]
> Then again I think it is overridden in the CMakeLists.txt, so it shouldn't break anything.

Looks a commit [June 12 2023](https://github.com/sccn/liblsl/commit/dfc81db275e725ea1329d02987ab88ec0c0c35b5) removed a conditional based on MSVC_VERSION that probably changed it from being set only on Windows to having it be set on all systems, and 1.16.2 came out [May 27, 2023](https://github.com/sccn/liblsl/releases/tag/v1.16.2) -- that could explain why we need to set something here, and suggests it could be removed once `>1.16.2` lands.

But in any case yes one of you two just needs to click the green button now!

**Pull request merged**